### ISSUE TITLE
ParamFetcher: return empty array if no params

### ISSUE THREAD

**Issue by NikitaKharkov** [FriendsOfSymfony/FOSRestBundle#1685]
The standard `Request` return always only that parameters which comes. I think need to improve some param which returns only existed parameters in query.
In my case I have to always check the values. 

```
foreach ($fetcher->all(true) as $param => $value) {
       if ($value !== null) {
               $paramsFetcher[$param] = $value
       }
}
// no parameters - no result
if ($paramsFetcher !== []) {
     $customers = $this->get('app.customer_manager')->getCustomerBySearchParams($paramsFetcher);
}
```
I read all the documentation and deep into the core and didn't find anything approaching to case. Can somebody suggest right decision?

**Comment by Tobion** [comment]
I guess the problem is that the param default is `null` automatically. So all params return null by default which is also why you get all them all when fetching all params.
Maybe it should be possible to set no default value (not even null) on a param. When you fetch it and no value is set, it throw an exception. When you fetch all params, you don't get the ones without a default value.

**Comment by NikitaKharkov** [comment]
Exactly! So I want to provide some flag "noDefault" to not init the key. In $fetcher array
